### PR TITLE
CFN list transform

### PR DESCRIPTION
**Pull request by tthyer**
adding a list transform macro for cloud formation

### CODE REVIEW

**Comment by tthyer** [comment]
@zaro0508 Please review

File: lambdas/cfn-list-transform-macro/list_transform/app.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import json
+import traceback
+
+
+def handle_transform(parameters):
+  operation = parameters['Operation']
+  transform_string = parameters['TransformString']
+  fragment = []
+  input_list = parameters['List']
```
Comment (zaro0508) [inline lambdas/cfn-list-transform-macro/list_transform/app.py]: should this be `parameters['InputList']`?

also maybe you want to check that it's actually a list?

File: lambdas/cfn-list-transform-macro/list_transform/app.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+def handler(event, context):
+    status = 'success'
+    response = {
+      'requestId': event['requestId'],
+      'status': 'success'
+    }
+
+    try:
+      response['fragment'] = handle_transform(event['params'])
```
Comment (zaro0508) [inline lambdas/cfn-list-transform-macro/list_transform/app.py]: I believe the parameters are in a `templateParameterValues` key, so this should be something like..
```
parameters = event["templateParameterValues"]
fragment = handle_transform(parameters)
```

**Comment by tthyer** [comment]
@zaro0508 Thanks for the feedback. Before I address all these comments, we need to discuss whether this strategy is going to work at all. It appears it won't because one can only use the SecurityGroups in the AWS::EC2::Instance definition if those groups are defined on the default VPC. Since we are mostly using internalvpc and it is not the default, the solution to create all the security group names from the input ports is not going to work, unless you decide to make internalvpc the default vpc. Do you intend to do that?

File: lambdas/cfn-list-transform-macro/README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
+# cfn-list-transform-macro
```
[the remark this answers is not in the document]
Comment (tthyer) [reply lambdas/cfn-list-transform-macro/README.md]: added better explanation in 9e5e7cf

File: lambdas/cfn-list-transform-macro/tests/unit/expected_append.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
+{
```
[the remark this answers is not in the document]
Comment (tthyer) [reply lambdas/cfn-list-transform-macro/tests/unit/expected_append.json]: nope, removing

File: lambdas/cfn-list-transform-macro/tests/unit/test_handler.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    self.assertEqual(expected, fragment)
+
+
+  def test_missing_prepend(self):
+    with open(r'events/list_transform_append.json') as file:
+      test_template = json.load(file)
+    self.event = test_template
+    result = handler(self.event, None)
+    fragment = result['fragment']
+    expected = ['AlphaFoo', 'BetaFoo', 'GammaFoo']
```
[the remark this answers is not in the document]
Comment (tthyer) [reply lambdas/cfn-list-transform-macro/tests/unit/test_handler.py]: The point is to confirm that if one does not add the Prepend argument then the behavior is the same as including `Prepend: false`

File: lambdas/cfn-list-transform-macro/list_transform/app.py (hunk 1 above)
Comment (tthyer) [reply lambdas/cfn-list-transform-macro/list_transform/app.py]: No, it should not be `parameters['InputList']` but I will check that it's a list.

Comment (tthyer) [reply lambdas/cfn-list-transform-macro/list_transform/app.py]: check and test for check added in b20ab10

File: lambdas/cfn-list-transform-macro/list_transform/app.py (hunk 2 above)
Comment (tthyer) [reply lambdas/cfn-list-transform-macro/list_transform/app.py]: "params" is a sibling of "templateParameterValues" -- [reference](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-macros.html#template-macros-lambda-interface)

**Pull request closed without merging**